scrapy/http/request/__init__.py

```diff
@@ -174,19 +179,19 @@ def __repr__(self) -> str:
     def copy(self) -> "Request":
         return self.replace()
 
-    def replace(self, *args, **kwargs) -> "Request":
+    def replace(self, *args: Any, **kwargs: Any) -> "Request":
```

Comment:
I think technically the return type is different; the same applies to .copy() - the class is preserved by default, unless "cls" attribute is passed. This could matter for classes like FormRequest or other custom request classes. Not a big deal, but I wonder if having self type as a default return type is better than hardcoding Request? We can also try to support `cls` typing directly, though it's more work.

Reply:
Hm, was thinking that Request typing is more restrictive than Self typing, for the user code. 

We're telling that the result of .replace (and other similar methods) no longer has all the new methods or attributes of the Request subclass, so mypy is going to show an error if user tries to access them.

Reply:
Neither Request nor Self is 100% correct here, but I wonder if using Self is going to cause fewer false positives in the type checks for the users.

Reply:
> Without the cls edge case, Self typing looks correct

Yup.

We could add an overload to all these methods that covers the case when `cls is not None`, though that requires adding a `cls` named arg explicitly, not sure if that will break anything.

Reply:
I think adding cls explicitly should be fine.